Clearing existing directory fails if directory is already empty during `make build-client-<foo>`

Steps taken:
- Cloned the repo
- Ran `make setup-new-sdk`
- Filled in the `config.overrides.json`
- Ran `make build-client-rust`, ended up with this error:
```
% make build-client-rust
make build-client sdk_language=rust tmpdir=/tmp/tmp.wm6Z2
make[1]: Entering directory '/home/nett/Stuff/openfga-sdk-generator'
mkdir -p "/home/nett/Stuff/openfga-sdk-generator/docs/openapi/" "/home/nett/Stuff/openfga-sdk-generator/clients/"
mkdir -p "/home/nett/Stuff/openfga-sdk-generator/docs/openapi"
curl "https://raw.githubusercontent.com/openfga/api/main/docs/openapiv2/apidocs.swagger.json" \
     -o "/home/nett/Stuff/openfga-sdk-generator/docs/openapi/openfga.openapiv2.raw.json"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 63399  100 63399    0     0   435k      0 --:--:-- --:--:-- --:--:--  439k
cat "/home/nett/Stuff/openfga-sdk-generator/docs/openapi/openfga.openapiv2.raw.json" | \
	docker run --rm -i stedolan/jq \
	 '(.. | .tags? | select(.)) |= ["OpenFga"] | (.tags? | select(.)) |= [{"name":"OpenFga"}] | del(.definitions.ReadTuplesParams, .definitions.ReadTuplesResponse, .paths."/stores/{store_id}/read-tuples", .definitions.StreamedListObjectsRequest, .definitions.StreamedListObjectsResponse, .paths."/stores/{store_id}/streamed-list-objects")' > \
	/home/nett/Stuff/openfga-sdk-generator/docs/openapi/openfga.openapiv2.json
sed -i -e 's/v1.//g' /home/nett/Stuff/openfga-sdk-generator/docs/openapi/openfga.openapiv2.json
# Build a config from common + overrides
jq -rs 'reduce .[] as $item ({}; . * $item)' "/home/nett/Stuff/openfga-sdk-generator/config/common/config.base.json" "/home/nett/Stuff/openfga-sdk-generator/config/clients/rust/config.overrides.json" > "/tmp/tmp.wm6Z2/config.json"
mkdir -p "/home/nett/Stuff/openfga-sdk-generator/clients/fga-rust-sdk"
# Initialize the temporary template directory
mkdir "/tmp/tmp.wm6Z2/template"
# Copy the shared files into temp template
cp -r "/home/nett/Stuff/openfga-sdk-generator/config/common/files/." "/tmp/tmp.wm6Z2/template/"
# Copy the template files into temp template
cp -r "/home/nett/Stuff/openfga-sdk-generator/config/clients/rust/template/." "/tmp/tmp.wm6Z2/template/"
# Copy the CHANGELOG.md file into temp template
cp "/home/nett/Stuff/openfga-sdk-generator/config/clients/rust/CHANGELOG.md.mustache" "/tmp/tmp.wm6Z2/template/"
# Clear existing directory
cd "/home/nett/Stuff/openfga-sdk-generator/clients/fga-rust-sdk" && ls -A | grep -Ev '.git|node_modules|.idea' | xargs rm -r && cd -
rm: missing operand
Try 'rm --help' for more information.
make[1]: *** [Makefile:149: build-client] Error 123
make[1]: Leaving directory '/home/nett/Stuff/openfga-sdk-generator'
make: *** [Makefile:245: build-client-rust] Error 2
```
This seems to be related to this SO question: https://stackoverflow.com/questions/36617999/error-rm-missing-operand-when-using-along-with-find-command
`cd "/home/nett/Stuff/openfga-sdk-generator/clients/fga-rust-sdk" && ls -A | grep -Ev '.git|node_modules|.idea'` returns exit code 1 as the directory is completely empty:
```
% ls -A clients/fga-rust-sdk
% cd "/home/nett/Stuff/openfga-sdk-generator/clients/fga-rust-sdk" && ls -A | grep -Ev '.git|node_modules|.idea'
% echo $?
1
```

--- a/scripts/build_client.sh
+++ b/scripts/build_client.sh
@@ -30,7 +30,7 @@ cp "${CONFIG_DIR}/clients/${SDK_LANGUAGE}/CHANGELOG.md.mustache" "${TMP_DIR}/tem
 
 # Clear existing directory
 # shellcheck disable=SC2010
-cd "${CLIENTS_OUTPUT_DIR}/fga-${SDK_LANGUAGE}-sdk" && ls -A | grep -Ev '.git|node_modules|.idea|venv|.gradle' | xargs rm -r || cd -
+cd "${CLIENTS_OUTPUT_DIR}/fga-${SDK_LANGUAGE}-sdk" && ls -A | { grep -Ev '.git|node_modules|.idea|venv|.gradle' || test $? = 1; } | xargs rm -r && cd -
 
 # Copy the generator ignore file into target directory (we need to do this before build otherwise openapi-generator ignores it)
 cp "${CONFIG_DIR}/clients/${SDK_LANGUAGE}/.openapi-generator-ignore" "${CLIENTS_OUTPUT_DIR}/fga-${SDK_LANGUAGE}-sdk/"
@@ -48,6 +48,7 @@ docker run --rm \
   "openapitools/openapi-generator-cli:${OPENAPI_GENERATOR_CLI_DOCKER_TAG}" generate \
   -i /docs/openapi/openfga.openapiv2.json \
   --http-user-agent="$http_user_agent" \
+  "${library[@]}" \
   -o "/clients/fga-${SDK_LANGUAGE}-sdk" \
   -c /config/config.json \
   -g "$(cat ./config/clients/"${SDK_LANGUAGE}"/generator.txt)"

--- a/scripts/build_client.sh
+++ b/scripts/build_client.sh
@@ -30,7 +30,7 @@ cp "${CONFIG_DIR}/clients/${SDK_LANGUAGE}/CHANGELOG.md.mustache" "${TMP_DIR}/tem
 
 # Clear existing directory
 # shellcheck disable=SC2010
-cd "${CLIENTS_OUTPUT_DIR}/fga-${SDK_LANGUAGE}-sdk" && ls -A | grep -Ev '.git|node_modules|.idea|venv|.gradle' | xargs rm -r && cd -
+cd "${CLIENTS_OUTPUT_DIR}/fga-${SDK_LANGUAGE}-sdk" && ls -A | grep -Ev '.git|node_modules|.idea|venv|.gradle' | xargs rm -r || cd -
 
 # Copy the generator ignore file into target directory (we need to do this before build otherwise openapi-generator ignores it)
 cp "${CONFIG_DIR}/clients/${SDK_LANGUAGE}/.openapi-generator-ignore" "${CLIENTS_OUTPUT_DIR}/fga-${SDK_LANGUAGE}-sdk/"
@@ -48,7 +48,6 @@ docker run --rm \
   "openapitools/openapi-generator-cli:${OPENAPI_GENERATOR_CLI_DOCKER_TAG}" generate \
   -i /docs/openapi/openfga.openapiv2.json \
   --http-user-agent="$http_user_agent" \
-  "${library[@]}" \
   -o "/clients/fga-${SDK_LANGUAGE}-sdk" \
   -c /config/config.json \
   -g "$(cat ./config/clients/"${SDK_LANGUAGE}"/generator.txt)"

--- a/scripts/build_client.sh
+++ b/scripts/build_client.sh
@@ -29,7 +29,7 @@ cp -r "${CONFIG_DIR}/clients/${SDK_LANGUAGE}/template/." "${TMP_DIR}/template/"
 cp "${CONFIG_DIR}/clients/${SDK_LANGUAGE}/CHANGELOG.md.mustache" "${TMP_DIR}/template/"
 
 # Clear existing directory
-# shellcheck disable=SC2010
+# shellcheck disable=SC2010,SC2012
 cd "${CLIENTS_OUTPUT_DIR}/fga-${SDK_LANGUAGE}-sdk" && ls -A | { grep -Ev '.git|node_modules|.idea|venv|.gradle' || test $? = 1; } | xargs rm -r && cd -
 
 # Copy the generator ignore file into target directory (we need to do this before build otherwise openapi-generator ignores it)

